pkg/cmd/cli/describe/deployments.go

```diff
@@ -93,19 +93,30 @@ func (d *DeploymentConfigDescriber) Describe(namespace, name string, settings kc
 			if err == nil {
 				sorted := deploymentsHistory.Items
 				sort.Sort(sort.Reverse(rcutils.OverlappingControllers(sorted)))
+				counter := 0
 				for _, item := range sorted {
 					if item.Name != deploymentName && deploymentConfig.Name == deployutil.DeploymentConfigNameFor(&item) {
 						header := fmt.Sprintf("Deployment #%d", deployutil.DeploymentVersionFor(&item))
 						printDeploymentRc(&item, d.kubeClient, out, header, false)
+						counter++
+					}
+					// Only show last three deployments here
+					if counter == 2 {
+						break
 					}
 				}
 			}
 		}
 
 		// Events
 		if events, err := d.kubeClient.Events(deploymentConfig.Namespace).Search(deploymentConfig); err == nil && events != nil {
+			latestDeploymentEvents := &kapi.EventList{Items: []kapi.Event{}}
+			// Only show last 5 events here
```

Comment:
Showing events for the previous deployment is not worse that what we are doing atm so a bigger number that accomodates all standard events for a deployment (2 for each hook -we could possibly identify how many hooks we have- one for the created deployment, one that it successfully deployed or failed - 8 or 10 in total) is acceptable. The gist is that we should be more sophisticated here in the way we choose the number of events to show.